storage/rocksdb/rdb_datadic.cc

```diff
@@ -940,9 +1067,17 @@ uint Rdb_key_def::pack_record(const TABLE *const tbl, uchar *const pack_buffer,
     // Primary Keys are special: for them, store the unpack_info even if it's
     // empty (provided m_maybe_unpack_info==true, see
     // ha_rocksdb::convert_record_to_storage_format)
-    if (len == RDB_UNPACK_HEADER_SIZE &&
-        m_index_type != Rdb_key_def::INDEX_TYPE_PRIMARY) {
-      unpack_info->clear();
+    if (m_index_type == Rdb_key_def::INDEX_TYPE_SECONDARY) {
+      size_t empty_unpack_info_len = RDB_UNPACK_HEADER_SIZE;
+      if (use_covered_bitmap_format()) {
+        empty_unpack_info_len += RDB_COVERED_BITMAP_SIZE;
+      }
+      if (len == empty_unpack_info_len && covered_bits == 0) {
+        unpack_info->clear();
+      } else if (use_covered_bitmap_format()) {
+        unpack_info->write_uint16_at(covered_bitmap_pos, covered_bits);
+        unpack_info->write_uint8_at(0, RDB_UNPACK_COVERED_DATA_TAG);
```

Comment:
I think this code which setups up the unpack info header should go up where the other data tag is written. It would be easier to reason about the code if the code that handles RDB_UNPACK_DATA_TAG is grouped together, and code that handles RDB_UNPACK_COVERED_DATA_TAG is grouped together. This way, if we added a third type, etc., this function can be more easily amended to include it.

This step of the process is mainly recording the length of the total unpack_info for secondary keys and if it's a covered index, recording the covered index bitmap along with the length.